compute_tools/src/migration.rs

```diff
@@ -81,53 +81,48 @@ impl<'m> MigrationRunner<'m> {
-        while current_migration < self.migrations.len() {
-            macro_rules! migration_id {
-                ($cm:expr) => {
-                    ($cm + 1) as i64
-                };
-            }
+            // Even though we are skipping the migration, updating the
+            // migration ID should help keep logic easy to understand when
+            // trying to understand the state of a cluster.
+            self.update_migration_id(migration_id)?;
+        } else {
+            info!("Running migration id={}:\n{}\n", migration_id, migration);
 
-            let migration = self.migrations[current_migration];
+            self.client
+                .simple_query("BEGIN")
+                .context("begin migration")?;
 
-            if migration.starts_with("-- SKIP") {
-                info!("Skipping migration id={}", migration_id!(current_migration));
+            self.client
+                .simple_query(migration)
+                .with_context(|| format!("run_migrations migration id={migration_id}"))?;
 
-                // Even though we are skipping the migration, updating the
-                // migration ID should help keep logic easy to understand when
-                // trying to understand the state of a cluster.
-                self.update_migration_id(migration_id!(current_migration))?;
-            } else {
-                info!(
-                    "Running migration id={}:\n{}\n",
-                    migration_id!(current_migration),
-                    migration
-                );
+            self.update_migration_id(migration_id)?;
 
-                self.client
-                    .simple_query("BEGIN")
-                    .context("begin migration")?;
+            self.client
+                .simple_query("COMMIT")
+                .context("commit migration")?;
 
-                self.client.simple_query(migration).with_context(|| {
-                    format!(
-                        "run_migrations migration id={}",
-                        migration_id!(current_migration)
-                    )
-                })?;
+            info!("Finished migration id={}", migration_id);
+        }
 
-                self.update_migration_id(migration_id!(current_migration))?;
+        Ok(())
+    }
 
-                self.client
-                    .simple_query("COMMIT")
-                    .context("commit migration")?;
+    /// Run the configrured set of migrations
+    pub fn run_migrations(mut self) -> Result<()> {
+        self.prepare_database()?;
 
-                info!("Finished migration id={}", migration_id!(current_migration));
-            }
+        let mut current_migration = self.get_migration_id()? as usize;
+        while current_migration < self.migrations.len() {
+            self.run_migration(
+                (current_migration + 1) as i64,
+                self.migrations[current_migration],
+            )?;
```

Comment:
NIT: Maybe that's only me, but current helper and indexes usage confuse me a bit

`run_migration()` has access to `self` so to `self.migrations` as well, which means we don't need to pass migration, only id, and can simplify it to

```suggestion
        let last_migration_id = self.get_migration_id()? as usize;
        let mut current_migration_id = last_migration_id + 1
        while current_migration_id < self.migrations.len() {
            self.run_migration(
                current_migration_id as i64,
            )?;
            
            current_migration_id += 1;
```
Then do `self.migrations[migration_id - 1]` inside `run_migration()` (just maybe add an extra sanity check `< self.migrations.len()` there too). Also made indexes to be more explicit.

WDYT?

Reply:
Good feedback; let me know what you think now.

Reply:
Actually, given https://github.com/neondatabase/neon/pull/10232#issuecomment-2573320536, to satisfy the borrow checker, I need to remove access to `self` in `run_migration`. Let me know what you think.